app/bundles/EmailBundle/Controller/AjaxController.php

```diff
@@ -229,14 +229,14 @@ protected function testEmailServerConnectionAction(Request $request)
             }
 
             if (!empty($mailer)) {
-                if (is_callable([$mailer, 'setApiKey'])) {
+                if (method_exists($mailer, 'setApiKey')) {
                     if (empty($settings['api_key'])) {
                         $settings['api_key'] = $this->get('mautic.helper.core_parameters')->getParameter('mailer_api_key');
                     }
                     $mailer->setApiKey($settings['api_key']);
                 }
 
-                if (is_callable([$mailer, 'setUsername']) && is_callable([$mailer, 'setPassword'])) {
+                if (method_exists($mailer, 'setUsername') && method_exists($mailer, 'setPassword')) {
```

Comment:
Right - that's what the PR I referenced is fixing. method_exists doesn't work for SMTP because the method doesn't exist since Swiftmailer's transport uses __call for those.